mlxbf-bootctl.c

```diff
@@ -539,20 +651,15 @@ void write_bootstream(const char *bootstream, const char *bootfile, int flags)
   }
 
   // Copy the bootstream to the bootfile device
-  int ifd = open(bootstream, O_RDONLY);
-  if (ifd < 0)
-    die("%s: %m", bootstream);
+  void *ibuf = malloc(MMC_BOOT_PARTITION_MAX_SIZE);
+  void *inidx = ibuf;
+  uint8_t padbuf[8] = {0}; // There are at most 8 pad bytes.
+  if (ibuf == NULL)
+    die("out of memory");
+  size_t bytes_left = read_bootstream_to_buffer(bootstream, ibuf, MMC_BOOT_PARTITION_MAX_SIZE, 1);
```

Comment:
why value 1 here, is it still working in progress?  Looks like version bitmap hasn;t been updated yet.

Reply:
Yes, it's still WIP. I'm adding a function to figure out the correct version to pass into that field.